File: verve patch/base/Applications/CSymphony_0_1/csSamples/CodeConstructor.cs
using System;
using System.CodeDom;
using System.CodeDom.Compiler;
using System.Collections;
using CSymphony.Lib;

namespace CSymphony.CSy
{
	public class CSyUnit : GraphUnit
	{
		public override void Build()
		{
			ns("MyNameSpace");

			// Class : BaseType
			declClass(TypePublic, "BaseType");

			declCtor(MemPublic);
			declCtor(MemPublic, "System.String", "TestStringParameter");

			// Class : DerviedType
			declClass(TypePublic, "DerivedType", "BaseType");
			ArgRef argTestParam = "TestStringParameter";

			declCtor(MemPublic, "System.String" , argTestParam);
			parent(argTestParam);

			declCtor(MemPublic);
			parent(new Literal("test"));

			declCtor(MemPublic, "System.Int32", "TestIntParameter");
			peer();

		}
	}
}

File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodeConstructor.vb
Option Strict On
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "BaseType")

			declCtor(MemPublic)
			
			declCtor(MemPublic, "System.String", "TestStringParameter")

			declClass(TypePublic, "DerivedType", "BaseType")
			
			Dim argTestParam As ArgRef  = New ArgRef("TestStringParameter")

			declCtor(MemPublic, "System.String" , argTestParam.Expr())
			parent(argTestParam.Expr())

			declCtor(MemPublic)
			parent(newLit("test"))

			declCtor(MemPublic, "System.Int32", "TestIntParameter")
			peer()
		End Sub
	End Class
End Namespace
			